structlog_sentry/__init__.py

```diff
@@ -1,19 +1,40 @@
 import logging
-import os
 import sys
+from typing import Optional, List, Tuple, Union
 
 from sentry_sdk import capture_event
+from sentry_sdk.integrations.logging import ignore_logger
 from sentry_sdk.utils import event_from_exception
 
 
 class SentryProcessor:
     """Sentry processor for structlog. Uses Sentry SDK to capture events in Sentry."""
 
-    def __init__(self, level=logging.WARNING, active=True):
+    def __init__(
+            self,
+            level: int = logging.WARNING,
+            active: bool = True,
+            as_extra: bool = True,
+            tag_keys: Union[List[str], str] = None) -> None:
+        """
+        :param level: events of this or higher levels will be reported to Sentry
+        :param active: a flag to make this processor enabled/disabled
+        :param as_extra: send `event_dict` as extra info to Sentry.
+        :param tag_keys: a list of keys. If any if these keys appear in `event_dict`,
+        the key and its corresponding value in `event_dict` will be used as Sentry event tags. use `"__all__"` to report
+        all key/value pairs of event as tags.
+        """
         self.level = level
         self.active = active
+        self.tag_keys = tag_keys
+        self._as_extra = as_extra
+        self._original_event_dict = None
 
-    def _log(self, event_dict, level):
+    def _get_event_and_hint(self, event_dict: dict) -> Tuple[dict, Optional[str]]:
+        """
+        create a sentry event and hint from structlog `event_dict` and sys.exc_info
```

Comment:
Please run `tox -e coala` and fix issues it reports.

(and I will also add a CI job for running coala)

Reply:
Done